node/src/main/scala/com/wavesplatform/state/BlockchainUpdaterImpl.scala

```diff
@@ -721,6 +721,8 @@ class BlockchainUpdaterImpl(blockchain: LevelDBWriter,
     }
   }
 
+  override def proofForBalanceOnHeight(address: Address, height: Height): Option[ProvenBalance] = blockchain.proofForBalanceOnHeight(address, height)
```

Comment:
This will not work for balance proofs at top height, is this by design?

Reply:
yes, because proof on current height doesn't make sense